tests/component_test.py

```diff
@@ -313,6 +317,143 @@ def run_test_relay(release_dir: pathlib.Path,
+    cmd = [str(release_dir / 'bin' / 'mailgun-relayery'),
+           '-channel_data_dir', channel_db_dir.as_posix(),
+           '-timestamps_data_dir', timestamp_db_dir.as_posix(),
+           '-api_key_dir', api_key_pth.as_posix(),
+           '-mailgun_address', 'http://127.0.0.1:{}'.format(port_mailgun),
+           '-address', ':{}'.format(port_relay)]
+    # yapf: enable
+
+    proc = subprocess.Popen(cmd, stdout=stdout)
+    with tests.proc.terminating(proc=proc, timeout=5):
+        # let the server initialize
+        tests.proc.sleep_while_process(proc=proc, seconds=1)
+
+        if proc.poll() is not None:
+            raise AssertionError("Expected the server process to be alive, but it died.")
+
+        url_rel = 'http://127.0.0.1:{}'.format(port_relay)
+        client_rel = tests.relay.RemoteCaller(url_rel)
+
+        message = tests.relay.Message(subject="a message from your friend", content="dear friend, I hope all is good.")
+
+        # error 429: too many requests
+        http_err = None  # type: Optional[requests.exceptions.HTTPError]
+        try:
+            _ = client_rel.put_message(descriptor=desc_large_min_period, token=token, message=message)
+            _ = client_rel.put_message(descriptor=desc_large_min_period, token=token, message=message)
+        except requests.exceptions.HTTPError as err:
+            http_err = err
+
+        # as query parameters can have any order, we need a list here. Will fix itself once we introduce header params.
+        expected_errs = ["429 Client Error: Too Many Requests for url: " \
+                         "{}/api/message?descriptor={}&token={}".format(url_rel, desc_large_min_period, token),
+                         "429 Client Error: Too Many Requests for url: " \
+                         "{}/api/message?token={}&descriptor={}".format(url_rel, token, desc_large_min_period)
+                         ]
+        assert http_err.__str__() in expected_errs, "expected one of {}, got {}".format(expected_errs, http_err)
+
+        # error 413: request entity too large
+        http_err = None  # type: Optional[requests.exceptions.HTTPError]
+        try:
+            _ = client_rel.put_message(descriptor=desc_small_max_size, token=token, message=message)
+        except requests.exceptions.HTTPError as err:
+            http_err = err
+
+        # as query parameters can have any order, we need a list here. Will fix itself once we introduce header params.
+        expected_errs = ["413 Client Error: Request Entity Too Large for url: " \
+                         "{}/api/message?descriptor={}&token={}".format(url_rel, desc_small_max_size, token),
+                         "413 Client Error: Request Entity Too Large for url: " \
+                         "{}/api/message?token={}&descriptor={}".format(url_rel, token, desc_small_max_size)
+                         ]
+        assert http_err.__str__() in expected_errs, "expected one of {}, got {}".format(expected_errs, http_err)
+
```

Comment:
add a test that updating the channel does not update the timestamps if min_period is the same.

also add it as a InTest contract -- the timestamp should be equal.

Reply:
I unfortunately can't add the condition that the timestamp did not change as gocontracts do not support a state; nor can I add it to the component test as the client has never direct access to the timestamps. I will add a unit test for this case.